sdk/src/client/mod.rs

```diff
@@ -58,4 +63,75 @@ impl Client {
             .await
             .map_err(Into::into)
     }
+
+    pub async fn broadcast(&self, actions: Vec<Action>) -> anyhow::Result<B::Output> {
+        let (block_hash, nonce) = self.get_nonce().await?;
+
+        let signed_tx = SignedTransaction::from_actions(
+            nonce,
+            self.signer.account_id.clone(),
+            self.engine.as_str().parse()?,
+            &self.signer.clone().into(),
+            actions,
+            block_hash,
+            0,
+        );
+
+        B::broadcast_tx(&self.client, signed_tx).await
+    }
+
+    pub async fn get_nonce(&self) -> anyhow::Result<(CryptoHash, u64)> {
+        let request = near_jsonrpc_primitives::types::query::RpcQueryRequest {
+            block_reference: near_primitives::types::Finality::Final.into(),
+            request: views::QueryRequest::ViewAccessKey {
+                account_id: self.signer.account_id.clone(),
+                public_key: self.signer.public_key.clone(),
+            },
+        };
+        let response = self.client.call(request).await?;
+        let block_hash = response.block_hash;
+        let nonce = match response.kind {
+            QueryResponseKind::AccessKey(k) => k.nonce + 1,
+            _ => anyhow::bail!("Wrong response kind: {:?}", response.kind),
+        };
+
+        Ok((block_hash, nonce))
+    }
+
+    pub fn with_engine(self, engine: AccountId) -> Self {
+        Self { engine, ..self }
+    }
+
+    pub fn with_signer(self, signer: InMemorySigner) -> Self {
+        Self { signer, ..self }
+    }
+
+    pub fn with_account_id(self, account_id: AccountId) -> Self {
+        Self {
+            signer: InMemorySigner {
+                account_id,
+                public_key: self.signer.public_key,
+                secret_key: self.signer.secret_key,
+            },
+            ..self
+        }
+    }
+
+    pub fn to_async(self) -> Client<broadcast::Async> {
```

Comment:
Usually, when we convert `Self` into something else and accept `self` by value, the convention to name such methods with `into_*`. prefix.